cea/interfaces/dashboard/map_layers/base.py

```
@@ -45,39 +48,98 @@ def expected_parameters(self) -> dict:
+        for locator_descriptor in self.input_file_locators:
+            locator_callable = locator_descriptor
+            if len(locator_descriptor) > 1:
+                func, args = locator_descriptor
 
-        def generate_file_list():
-            for locator_descriptor in self.input_files:
-                locator_callable = locator_descriptor
-                if len(locator_descriptor) > 1:
-                    func, args = locator_descriptor
+                def _callable():
+                    return func(*args)
```

Ruff (0.7.0)

Function definition does not bind loop variable `func`

(B023)